packages/app-project/src/shared/components/CollectionsModal/components/SelectCollection/SelectCollection.js

```diff
@@ -1,19 +1,29 @@
 import { Box, Button, FormField, Grid, Select } from 'grommet'
 import PropTypes from 'prop-types'
 import { useTranslation } from 'next-i18next'
+import { useState } from 'react'
 
 function SelectCollection ({
-  collections,
-  disabled,
+  collections = [],
+  disabled = false,
   onSelect,
-  onSearch,
   onSubmit,
   selected
 }) {
   const { t } = useTranslation('components')
+  const [options, setOptions] = useState(collections)
+
+  /** https://storybook.grommet.io/?path=/story/input-select-search--search */
+  const onSearch = (text) => {
+    const escapedText = text.replace(/[-\\^$*+?.()|[\]{}]/g, '\\$&')
+    const exp = new RegExp(escapedText, 'i')
+    setOptions(collections.filter((collection) => exp.test(collection.display_name)))
+  }
```

Comment:
This won't work, unfortunately. It filters the first page of collections returned from Panoptes, but a volunteer can be a collaborator on more than one page of collections. 

You need to query Panoptes using the collections search API, and show the first page of results.

Reply:
The API query should look like this:

```js
{
  favorite: false,
  current_user_roles: 'owner,collaborator,contributor',
  search: text
}
```